Ports/.port_include.sh

```diff
@@ -41,7 +41,8 @@ target_env() {
 
 target_env
 
-DESTDIR="${SERENITY_INSTALL_ROOT}"
+SERENITY_ARCH="${SERENITY_ARCH:-$(uname -m)}"
+DESTDIR="${SERENITY_INSTALL_ROOT:-}"
```

Comment:
Will these even work, considering that we just sourced `.hosted_defs.sh` in the previous line?

In fact, both of these settings should just be in a special case inside `.hosted_defs.sh`.

Reply:
`.hosted_defs.sh` isn't copied to serenity. it contains settings that you only want when cross compiling.

Reply:
Could we at least do something like the following then? Setting the correct settings for on-Serenity builds via fallbacks _somewhere_ in the script doesn't feel quite right.

```diff
diff --git a/Ports/.port_include.sh b/Ports/.port_include.sh
index bf9e58a009..03d5844340 100755
--- a/Ports/.port_include.sh
+++ b/Ports/.port_include.sh
@@ -29,14 +29,16 @@ buildstep_intro() {
     echo -e "\x1b[1;32m=> $@\x1b[0m"
 }
 
-maybe_source() {
-    if [ -f "$1" ]; then
-        . "$1"
-    fi
-}
-
 target_env() {
-    maybe_source "${SCRIPT}/.hosted_defs.sh"
+    if [ -f "${SCRIPT}/.hosted_defs.sh" ]; then
+        . "${SCRIPT}/.hosted_defs.sh"
+    elif [ "$(uname -s)" = "SerenityOS" ]; then
+        export SERENITY_ARCH="$(uname -m)"
+        export SERENITY_INSTALL_ROOT=""
+    else
+        >&2 echo "Error: .hosted_defs.sh is missing and we are not running on Serenity."
+        exit 1
+    fi
 }
 
 target_env
```

Reply:
We'd presumably also want to do a similar check for `host_env`, which relies on variables set by `.hosted_defs.sh`.

Reply:
done. host_env looks unused though.

Reply:
It is only used by ports that need to build on the host, for example `llvm`.